### ISSUE TITLE
Required 4443 and 10000-10100/udp

### ISSUE THREAD

**Issue by gurumark** [tiredofit/docker-jitsi-meet#4]
Hi,
I am having connection issues after opening up the required firewall ports in the host:
* Opened 4443 TCP
* Opened 32700-32900/udp ports in the host <--> 10000-10100/udp ports in the container
I get "You may need to check your network connection error" when I get into a room.

I hope I am making a rookie mistake and it is resolved quickly. 

Thanks for the repository and appreciate your help!

**Comment by tiredofit** [comment]
Jitsi meet is unneccessarily complex and I would recommend following the examples exactly to get most success. Otherwise, the /logs folder in the docker-compose will be very helpful to track down what is happening.

**Comment by gurumark** [comment]
Unfortunately, no luck here :-( 
I am using the docker-compose.xml in the examples directory -- modified the domain name. I have proxy set so I got my certs and everything ... can use the front page with https 80,443 works. 

However, when I get into a room it errors out. then I tried to change the port to:
 jvb:
    container_name: jvb
    image: tiredofit/jitsi-videobridge
    ports:
      - 4443:4443
      - 10000-10100:10000-10100/udp

To make it expose the correct ports in the host. Opened the firewall on UDP but still get the same error.

Here is my docker-compose log:

~/jitsi# docker-compose logs
Attaching to jvb, prosody, jicofo, jitsi-meet
prosody       | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
prosody       | [s6-init] ensuring user provided files have correct perms...exited 0.
prosody       | [fix-attrs.d] applying ownership & permissions fixes...
prosody       | [fix-attrs.d] 02-zabbix: applying... 
prosody       | [fix-attrs.d] 02-zabbix: exited 0.
prosody       | [fix-attrs.d] 03-logrotate: applying... 
prosody       | [fix-attrs.d] 03-logrotate: exited 0.
prosody       | [fix-attrs.d] done.
prosody       | [cont-init.d] executing container initialization scripts...
prosody       | [cont-init.d] 01-permissions: executing... 
prosody       | /var/run/s6/etc/cont-init.d/01-permissions: line 59: syntax error near unexpected token `fi'
prosody       | /var/run/s6/etc/cont-init.d/01-permissions: line 59: `  fi'
prosody       | [cont-init.d] 01-permissions: exited 2.
prosody       | [cont-init.d] 02-zabbix: executing... 
prosody       | [cont-init.d] 02-zabbix: exited 0.
prosody       | [cont-init.d] 03-cron: executing... 
prosody       | **** [cron] Enabling Cron
prosody       | [cont-init.d] 03-cron: exited 0.
prosody       | [cont-init.d] 04-smtp: executing... 
prosody       | **** [smtp] Disabling SMTP Features
prosody       | [cont-init.d] 04-smtp: exited 0.
prosody       | [cont-init.d] 10-prosody: executing... 
prosody       | Generating Certificates
prosody       | WARNING: ca-certificates.crt does not contain exactly one certificate or CRL: skipping
prosody       | [cont-init.d] 10-prosody: exited 0.
prosody       | [cont-init.d] 99-container-init: executing... 
prosody       | [cont-init.d] 99-container-init: exited 0.
prosody       | [cont-init.d] done.
prosody       | [services.d] starting services
prosody       | [services.d] done.
prosody       |  
prosody       | 
prosody       | ** [zabbix] Starting Zabbix Agent
prosody       | ** [prosody] Registering Focus User
prosody       | 
prosody       | ** [prosody] Starting prosody
prosody       | ** [cron] Starting cron
jicofo        | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
jicofo        | [s6-init] ensuring user provided files have correct perms...exited 0.
jicofo        | [fix-attrs.d] applying ownership & permissions fixes...
jicofo        | [fix-attrs.d] 02-zabbix: applying... 
jicofo        | [fix-attrs.d] 02-zabbix: exited 0.
jicofo        | [fix-attrs.d] 03-logrotate: applying... 
jicofo        | /var/run/s6/etc/cont-init.d/01-permissions: line 59: syntax error near unexpected token `fi'
jicofo        | /var/run/s6/etc/cont-init.d/01-permissions: line 59: `  fi'
jicofo        | [fix-attrs.d] 03-logrotate: exited 0.
jicofo        | [fix-attrs.d] done.
jicofo        | [cont-init.d] executing container initialization scripts...
jicofo        | [cont-init.d] 01-permissions: executing... 
jicofo        | [cont-init.d] 01-permissions: exited 2.
jicofo        | [cont-init.d] 02-zabbix: executing... 
jicofo        | [cont-init.d] 02-zabbix: exited 0.
jicofo        | [cont-init.d] 03-cron: executing... 
jicofo        | **** [cron] Enabling Cron
jicofo        | [cont-init.d] 03-cron: exited 0.
jicofo        | [cont-init.d] 04-smtp: executing... 
jicofo        | **** [smtp] Disabling SMTP Features
jicofo        | [cont-init.d] 04-smtp: exited 0.
jicofo        | [cont-init.d] 10-jicofo: executing... 
jicofo        | [cont-init.d] 10-jicofo: exited 0.
jicofo        | [cont-init.d] 99-container-init: executing... 
jicofo        | [cont-init.d] 99-container-init: exited 0.
jicofo        | [cont-init.d] done.
jicofo        | [services.d] starting services
jicofo        | [services.d] done.
jicofo        | 
jicofo        | ** [zabbix] Starting Zabbix Agent
jicofo        |  
jicofo        | ** [jicofo] Tue May  1 07:50:57 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:51:07 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:51:17 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:51:27 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:51:37 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:51:47 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [cron] Starting cron
jicofo        | ** [jicofo] Tue May  1 07:51:57 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:52:07 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:52:17 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:52:27 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:52:37 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:52:47 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:52:57 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | ** [jicofo] Tue May  1 07:53:07 PDT 2018 - Establishing connection with Prosody Host: prosody
jicofo        | 
jicofo        | ** [jicofo] Starting jicofo
jitsi-meet    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
jitsi-meet    | [s6-init] ensuring user provided files have correct perms...exited 0.
jitsi-meet    | [fix-attrs.d] applying ownership & permissions fixes...
jitsi-meet    | [fix-attrs.d] 02-zabbix: applying... 
jitsi-meet    | [fix-attrs.d] 02-zabbix: exited 0.
jitsi-meet    | [fix-attrs.d] 03-logrotate: applying... 
jitsi-meet    | [fix-attrs.d] 03-logrotate: exited 0.
jitsi-meet    | [fix-attrs.d] done.
jitsi-meet    | [cont-init.d] executing container initialization scripts...
jitsi-meet    | [cont-init.d] 01-permissions: executing... 
jitsi-meet    | **** [permissions] total 0 UIDs modified...
jitsi-meet    | **** [permissions] total 0 GIDs modified...
jitsi-meet    | [cont-init.d] 01-permissions: exited 0.
jitsi-meet    | [cont-init.d] 02-zabbix: executing... 
jitsi-meet    | [cont-init.d] 02-zabbix: exited 0.
jitsi-meet    | [cont-init.d] 03-cron: executing... 
jitsi-meet    | **** [cron] Enabling Cron
jitsi-meet    | [cont-init.d] 03-cron: exited 0.
jitsi-meet    | [cont-init.d] 04-smtp: executing... 
jitsi-meet    | **** [smtp] Disabling SMTP Features
jitsi-meet    | [cont-init.d] 04-smtp: exited 0.
jitsi-meet    | [cont-init.d] 10-nginx: executing... 
jitsi-meet    | [cont-init.d] 10-nginx: exited 0.
jitsi-meet    | [cont-init.d] 99-container-init: executing... 
jitsi-meet    | [cont-init.d] 99-container-init: exited 0.
jitsi-meet    | [cont-init.d] done.
jitsi-meet    | [services.d] starting services
jitsi-meet    | [services.d] done.
jitsi-meet    |  
jitsi-meet    | 
jitsi-meet    | ** [zabbix] Starting Zabbix Agent
jitsi-meet    | 
jitsi-meet    | ** [jitsi-meet] Starting nginx
jitsi-meet    | ** [cron] Starting cron
jvb           | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
jvb           | [s6-init] ensuring user provided files have correct perms...exited 0.
jvb           | [fix-attrs.d] applying ownership & permissions fixes...
jvb           | [fix-attrs.d] 02-zabbix: applying... 
jvb           | [fix-attrs.d] 02-zabbix: exited 0.
jvb           | [fix-attrs.d] 03-logrotate: applying... 
jvb           | [fix-attrs.d] 03-logrotate: exited 0.
jvb           | [fix-attrs.d] done.
jvb           | [cont-init.d] executing container initialization scripts...
jvb           | [cont-init.d] 01-permissions: executing... 
jvb           | /var/run/s6/etc/cont-init.d/01-permissions: line 59: syntax error near unexpected token `fi'
jvb           | /var/run/s6/etc/cont-init.d/01-permissions: line 59: `  fi'
jvb           | [cont-init.d] 01-permissions: exited 2.
jvb           | [cont-init.d] 02-zabbix: executing... 
jvb           | **** [zabbix] Disabling Zabbix
jvb           | [cont-init.d] 02-zabbix: exited 0.
jvb           | [cont-init.d] 03-cron: executing... 
jvb           | **** [cron] Enabling Cron
jvb           | [cont-init.d] 03-cron: exited 0.
jvb           | [cont-init.d] 04-smtp: executing... 
jvb           | **** [smtp] Disabling SMTP Features
jvb           | [cont-init.d] 04-smtp: exited 0.
jvb           | [cont-init.d] 10-jitsi-videobridge: executing... 
jvb           | [cont-init.d] 10-jitsi-videobridge: exited 0.
jvb           | [cont-init.d] 99-container-init: executing... 
jvb           | [cont-init.d] 99-container-init: exited 0.
jvb           | [cont-init.d] done.
jvb           | [services.d] starting services
jvb           | [services.d] done.
jvb           |  
jvb           | ** [jitsi-videobridge] Tue May  1 07:51:02 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:51:12 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:51:22 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:51:32 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:51:42 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:51:52 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [cron] Starting cron
jvb           | ** [jitsi-videobridge] Tue May  1 07:52:02 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:52:12 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:52:23 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:52:33 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:52:43 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:52:53 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:53:03 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | ** [jitsi-videobridge] Tue May  1 07:53:13 PDT 2018 - Establishing connection with Prosody Host: prosody
jvb           | 
jvb           | ** [jitsi-videobridge] Starting jitsi-videobridge

*****
I was hoping this will work without spending countless hours. 
I will wait for your response before giving up.

**Comment by tiredofit** [comment]
I see an error in the prosody startup script. Will look into in a few hours - Haven't seen that before. As for Jitsi - Yup - its a few hour process to get right as per experience and hearing from other people....

**Comment by tiredofit** [comment]
I also see it in Jicofo too. Hang in there.

**Comment by gurumark** [comment]
Is there any news on these?

**Comment by tiredofit** [comment]
Sorry, looking closer at it, the issues that are coming out are not related to Jitsi or Jicofo or Prosody, they are related to my base image which allows for on the fly UID/GID Changes. I've solved it on my private repo and will sanitize and push them up to Docker hub, although it won't affect/fix the issue you are seeing.

**Comment by gurumark** [comment]
Thanks for your time. 
Do you know any other jitsi repository that works following the instructions?

**Comment by tiredofit** [comment]
Not to toot my image's horn, but I'm going to say that this is the most recent, working solution.
As with anything that relies on heavy networking in Docker, there is going to be a learning curve. There are a few other complicated images that I have built in the same fashion that definitely take some work and knowledge. As you might have saw, the Jitsi Meet documentation is there, but lacking. 

That being said, I'm available for consultancy for bringing this to work in production environments, I am simply sharing these images to the community on essentially an as is basis sanitized from my own private organization repos.

**Comment by gurumark** [comment]
Not to toot my image's horn, but I'm going to say that this is the most recent, working solution.
-- I followed your instructions to the letter and could not get it working. 

It should work when you run the example docker-compose.xml without any modifications and spending hours. I shared my compose.xml file as well. It is docker so it is not really your box vs my box. It's got to work when you turn it on. If it's not working then well... it is not really "working solution". Isn't it?

I understand this is a free rep and you are kind enough to share it with public but claiming it is a "working solution" does not really represent the current state of your rep. If you fix the issues, I will be glad to say it is a "working solution" but unfortunately, after wasting my 2 hours, I cannot say it is.
